span-processing-utils/src/main/java/org/hypertrace/config/span/processing/utils/SpanFilterMatcher.java

```diff
@@ -111,7 +111,13 @@ private boolean matches(String lhs, String rhs, RelationalOperator relationalOpe
       case RELATIONAL_OPERATOR_ENDS_WITH:
         return lhs.endsWith(rhs);
       case RELATIONAL_OPERATOR_REGEX_MATCH:
-        return Pattern.compile(rhs).matcher(lhs).find();
+        try {
+          return Pattern.compile(rhs).matcher(lhs).find();
+        } catch (Exception e) {
+          log.error("Invalid regex: {} passed to match: {}", rhs, e);
+          log.debug("Invalid regex passed to match. Hence returning false. lhs: {} and rhs: {}", lhs, rhs);
```

Comment:
```suggestion
       if (log.isDebugEnabled()) {
          log.debug("Invalid regex passed to match. Hence returning false. lhs: {} and rhs: {}", lhs, rhs);
        }
```

Reply:
Oh! log automatically doesn't take care of whether debug is enabled or not?

Reply:
It does, but optimises on performance if you are creating heavy strings in the log message

https://stackoverflow.com/questions/963492/in-log4j-does-checking-isdebugenabled-before-logging-improve-performance